rpc/types_transaction.go

```diff
@@ -89,6 +109,77 @@ type DeclareTxnV2 struct {
 	ClassHash         *felt.Felt         `json:"class_hash"`
 }
 
+type DeclareTxnV3 struct {
+	Type              TransactionType       `json:"type"`
+	SenderAddress     *felt.Felt            `json:"sender_address"`
+	CompiledClassHash *felt.Felt            `json:"compiled_class_hash"`
+	Version           TransactionVersion    `json:"version"`
+	Signature         []*felt.Felt          `json:"signature"`
+	Nonce             *felt.Felt            `json:"nonce"`
+	ClassHash         *felt.Felt            `json:"class_hash"`
+	ResourceBounds    ResourceBoundsMapping `json:"resource_bounds"`
+	Tip               *felt.Felt            `json:"tip"`
+	// The data needed to allow the paymaster to pay for the transaction in native tokens
+	PayMasterData []*felt.Felt `json:"paymaster_data"`
+	// The data needed to deploy the account contract from which this tx will be initiated
+	AccountDeploymentData []*felt.Felt `json:"account_deployment_data"`
+	// The storage domain of the account's nonce (an account has a nonce per DA mode)
+	NonceDataMode DAMode `json:"nonce_data_availability_mode"`
+	// The storage domain of the account's balance from which fee will be charged
+	FeeMode DAMode `json:"fee_data_availability_mode"`
+}
+
+type ResourceBoundsMapping struct {
+	// The max amount and max price per unit of L1 gas used in this tx
+	L1Gas ResourceBounds `json:"l1_gas"`
+	// The max amount and max price per unit of L2 gas used in this tx
+	L2Gas ResourceBounds `json:"l2_gas"`
+}
+
+type DAMode string
+
+const (
+	DAModeL1 DAMode = "L1"
+	DAModeL2 DAMode = "L2"
+)
+
+func (da *DAMode) UInt64() (uint64, error) {
+	switch *da {
+	case DAModeL1:
+		return uint64(0), nil
+	case DAModeL2:
+		return uint64(1), nil
+	}
+	return 0, errors.New("Unknown DAMode")
+}
+
+type Resource string
+
+const (
+	ResourceL1Gas Resource = "L1_GAS"
+	ResourceL2Gas Resource = "L2_GAS"
+)
+
+type ResourceBounds struct {
+	// The max amount of the resource that can be used in the tx
+	MaxAmount *felt.Felt `json:"max_amount"`
+	// The max price per unit of this resource for this tx
+	MaxPricePerUnit *felt.Felt `json:"max_price_per_unit"`
+}
+
+func (rb ResourceBounds) Bytes(resource Resource) []byte {
+	const eight = 8
+	maxAmountBytes := make([]byte, eight)
+	binary.BigEndian.PutUint64(maxAmountBytes, rb.MaxAmount.Impl().Uint64())
+	maxPriceBytes := rb.MaxPricePerUnit.Bytes()
+	return utils.Flatten(
+		[]byte{0},
+		[]byte(resource),
+		maxAmountBytes,
+		maxPriceBytes[16:], // Last 16 bytes of the uint128.
```

Comment:
16 bytes is a `uint128`.
```suggestion
		maxPriceBytes[16:], // uint128.
```

I fixed this comment in Juno.